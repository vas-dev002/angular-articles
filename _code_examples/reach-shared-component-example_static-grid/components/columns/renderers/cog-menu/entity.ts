import { ICellRendererParams } from "@ag-grid-community/core";

export interface ICogMenuEvent {
	params: ICellRendererParams;
	rowIndex: number;
}
export interface ICogMenuItem {
	css?: string;
	hide?: (event: ICogMenuEvent) => boolean;
	name: string;
	onClick: (event: ICogMenuEvent) => void;
}
export interface ICellCogMenuParams extends ICellRendererParams {
	menuItems: ICogMenuItem[];
	isDisabled?: (cellData: unknown) => boolean;
	getCogMenuTooltip?: (cellData: unknown) => string;
}
