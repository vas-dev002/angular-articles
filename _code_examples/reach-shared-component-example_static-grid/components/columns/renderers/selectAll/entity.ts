import { IHeaderParams } from "@ag-grid-community/core";
import { EventEmitter } from "@angular/core";

export interface ISelectAllMethods {
	selectAll: () => void;
	deselectAll: () => void;
	getRowsCount: (excludeDisabledRows: boolean) => number;
	getSelectedCount: () => number;
	onSelectionChanged: EventEmitter<Object[]>;
}

export interface ISelectAllParams extends IHeaderParams {
	suppressHeaderCheckBox?: boolean;
	selectMethods: ISelectAllMethods;
}
