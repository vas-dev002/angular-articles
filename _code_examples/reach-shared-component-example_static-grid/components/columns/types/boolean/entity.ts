export interface BooleanGroupParams {
	value: string;
}
