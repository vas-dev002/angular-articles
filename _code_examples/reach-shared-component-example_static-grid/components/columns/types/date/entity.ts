import { ValueFormatterParams } from "@ag-grid-community/core";

export interface IDateValue {
	date: string;
}

export interface IDateGroupKeyParams {
	value: string;
	date: string;
}

export interface IDateValueFormatterParams extends ValueFormatterParams {
	value: string | IDateValue;
}
