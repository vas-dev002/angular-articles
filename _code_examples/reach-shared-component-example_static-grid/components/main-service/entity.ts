import { StaticGridService } from "./static-grid.service";
import { Dictionary } from "./dictionary";

export class ServiceSGDictionary extends Dictionary<string, StaticGridService> { }
