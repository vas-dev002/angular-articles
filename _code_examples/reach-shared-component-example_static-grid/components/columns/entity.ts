import { Type, Injector } from "@angular/core";
import { ICellRendererAngularComp } from "@ag-grid-community/angular";
import { ColDef } from "@ag-grid-community/core";
import { IColumnMetaSG } from "../main/entity";
import { StaticGridComponent } from "../main/comp";
import { ColumnDisplayType } from "./types";
import { IColDefAbstractFactory } from "./types/base/base-abstract-factory/models";
import { ObjectDetailsType } from "@evergreen-shared/entity";

export interface ITypeControllerSG {
	renderComponent: Type<ICellRendererAngularComp>;
	columnDefinition(): ColDef;
}

export interface IFilterOptionsSG {
	text: string;
	value: unknown;
	props?: Object;
	sortOrder?: number;
}

export interface ITypeControllerParams {
	meta: IColumnMetaSG;
	grid: StaticGridComponent;
	customOptions: Object;
	injector?: Injector;
}

export interface ILinkMeta {
	uRLColumnName: string;
	objectType: string;
	objectTypeColumnName: string;
	labelColumnName: string;
	keyColumnName: string;
	parameters: string;
	urlParameters: string[];
	renderLinkProperty: string;
}

export interface IOutcomeComparatorValue {
	status: number;
	text: string;
	tooltip: string;
}

export enum ApplicationForwardPathTypeEnum {
	None = 0,
	Retire = 1,
	Keep = 2,
	ForwardPath = 3,
	Uncategorized = 4
}

export enum ApplicationAssociationTypeEnum {
	None = 0,
	Dynamic = 1,
	StickyInclude = 2,
	StickyExclude = 3,
	ForwardPathed = 4,
	Mixed = 5
}

export enum StringValueGetterMode {
	String,
	SelectedItem,
	Params
}

export interface ITypedController {
	key: ColumnDisplayType;
	factory: Type<IColDefAbstractFactory>;
}

export enum TaskFlags {
	IsBucketTask = "IsBucketTask",
	IsReadOnlyTask = "IsReadOnlyTask",
	IsSystemTask = "IsSystemTask"
}

export interface IDateTimeEditorChanges {
	taskId: number;
	projectId: number;
	id: number;
	objectType: ObjectDetailsType;
	date: string;
	postBackParameters: { [key: string]: unknown };
}

export interface ICapacityDateTimeEditorChanges extends IDateTimeEditorChanges {
	slotId: number;
	slotName: string;
}
