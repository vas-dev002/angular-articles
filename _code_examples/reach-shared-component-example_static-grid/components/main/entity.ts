import { Type } from "@angular/core";
import {
	GetContextMenuItems,
	GetMainMenuItems,
	GridApi,
	RowNode,
	CellValueChangedEvent,
	ICellEditorParams
} from "@ag-grid-community/core";
import { ColumnState } from "@ag-grid-community/core/dist/cjs/columnController/columnController";
import { ICellRendererAngularComp, ICellEditorAngularComp } from "@ag-grid-community/angular";
import { IEditorTextValidator } from "../columns/editors/textEditor/text-editor.entity";
import { IColDefAbstractFactory } from "../columns/types/base/base-abstract-factory/models";

export interface IColumnMetaSG {
	columnName: string;
	lockVisible: boolean;
	visible: boolean;
	cssClass: string;
	headerName: string;
	controller: Type<IColDefAbstractFactory>;
	filterType: FilterTypeSG;
	isObjectKey: boolean;
	/** The temporary field until static grid has a connection with old version*/
	metaFromApi?: Object;
	/** The column name from data with tooltip*/
	tooltipField?: string;
	/** The column name from data with status of availability*/
	disabledField?: string;
}

export enum FilterTypeSG {
	Multiselect = "Multiselect",
	Number = "Number",
	String = "String",
	Date = "Date",
	DateTime = "DateTime",
	None = "None"
}

export interface IServiceColumn {
	serviceComponent: Type<ICellRendererAngularComp>;
	serviceParams: Object;
	index: number;
}

/** Configuration of Static Grid*/
export interface IOptionsSG {
	/** The message will be shown if data is not provided*/
	emptyMessage?: string;
	disableGroupBy?: boolean;
	editorsByKey?: { [columnKey: string]: IEditorOptions<{}> };
	serviceColumns?: IServiceColumn[];
	enableCheckBox?: boolean;
	suppressHeaderCheckBox?: boolean;
	dataKey?: DataKeyTypeSG;
	/** The object will be provided as a part of params for each column controller*/
	columnControllerOptions?: Object;
	exportFileName?: string;
	/** The options will be copied into agGrid options (GridOptions type)*/
	agGridOptions?: IAgGridOptions;
	styles?: Object;
	enableRowDragAndDrop?: boolean;
	suppressFilter?: boolean;
	suppressSortable?: boolean;
	suppressFloatingFilter?: boolean;
}

export interface IOptionsDynamicSG {
	dataUrl: string;
	agGridOptions?: IDynamicAgGridOptions;
}

export interface IShowingSelectionOptions {
	showCheckBoxes?: boolean;
	suppressHeaderCheckBox?: boolean;
}

export interface IDynamicAgGridOptions {
	cacheBlockSize?: number;
	maxBlocksInCache?: number;
	blockLoadDebounceMillis?: number;
}

export interface IAgGridOptions {
	getContextMenuItems?: GetContextMenuItems;
	getMainMenuItems?: GetMainMenuItems;
	localeTextFunc?: (key: string, defaultValue: string) => string;
	/** Define conditions for styles. Do not forget about ViewEncapsulation in Angular Components*/
	getRowClass?: (params: IGetRowClassParams<any>) => (string | string[]);
}

export interface IGetRowClassParams<TDataRow> {
	api: GridApi;
	data: TDataRow;
	node: RowNode;
	rowIndex: number;
}

export interface IGetCellClassParams<TDataRow, TCellValue> {
	value: TCellValue;
	api: GridApi;
	data: TDataRow;
	node: RowNode;
	rowIndex: number;
}

export interface IEditorOptions<Tvalidator> {
	editor: Type<ICellEditorAngularComp>;
	onChanged: (params: CellValueChangedEvent) => void;
	validator?: Tvalidator;
}

export interface ICellEditorParamsSG<Tvalidator> extends ICellEditorParams {
	validator?: Tvalidator;
}

export interface IEditorParams {
	meta: IColumnMetaSG;
	validator?: IEditorTextValidator;
}

export interface IInternalStateSG {
	groupColumns: string[];
	columns: ColumnState[];
	filters: ISelectedFilters;
	sortModel: ISortModel[];
}
export const defaultDataKeySG = "objectKey";
export type DataKeyTypeSG = string;
export type DataIdTypeSG = string | number;
export interface IDataRowSG {
	[key: string]: any;
}

export interface IFilterModel {
	type: string;
	filter: string;
	filterType: string;
}

export interface ISelectedFilters {
	[key: string]: IFilterModel;
}
export interface ISortModel {
	colId: string;
	sort: string;
}

export interface IDragAndDropModel {
	key: string | number;
	index: number;
}

export interface CellToUpdate {
	type: Type<ICellRendererAngularComp>;
	value: any;
}

export enum GridMode {
	StaticMode,
	DynamicMode,
	KeyValueMode
}

export enum RowModelType {
	ClientSide = "clientSide",
	Infinite = "infinite",
	Viewport = "viewport",
	ServerSide = "serverSide"
}
