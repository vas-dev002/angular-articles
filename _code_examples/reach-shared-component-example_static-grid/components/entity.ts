import { Observable } from "rxjs";
import { StaticGridComponent } from "./main/comp";

export interface IStaticGridWrapper {
	gridReady$: Observable<StaticGridComponent>;
}
