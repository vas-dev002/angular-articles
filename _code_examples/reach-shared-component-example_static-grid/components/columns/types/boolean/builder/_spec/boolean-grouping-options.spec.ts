import { BooleanGroupingOptions } from "..";
import { BooleanGroupParams } from "../../entity";

declare const expect: jest.Expect;

describe("[Builder option]: BooleanGroupingOptions", () => {
	let booleanGroupingOptions: BooleanGroupingOptions;

	beforeEach(() => {
		booleanGroupingOptions = new BooleanGroupingOptions();
	});

	describe("[BooleanGroupingOptions instantiate]", () => {
		it("should be successfully created", () => {
			expect(booleanGroupingOptions).toBeTruthy();
			expect(booleanGroupingOptions).toMatchSnapshot();
		});
	});

	describe("[Method]: getGroupKey", () => {
		it("should provide an unique grouping key", () => {
			const value1 = { value: "True" };
			const value2 = { value: "False" };

			const key1 = booleanGroupingOptions.getGroupKey(value1 as unknown as BooleanGroupParams);
			const key2 = booleanGroupingOptions.getGroupKey(value2 as unknown as BooleanGroupParams);

			expect(key1).toEqual(JSON.stringify("True"));
			expect(key2).toEqual(JSON.stringify("False"));
		});
	});
});
