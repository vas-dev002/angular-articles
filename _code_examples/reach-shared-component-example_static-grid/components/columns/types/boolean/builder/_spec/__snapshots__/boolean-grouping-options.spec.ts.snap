// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Builder option]: BooleanGroupingOptions [BooleanGroupingOptions instantiate] should be successfully created 1`] = `
BooleanGroupingOptions {
  "getGroupKey": [Function],
}
`;
